runtime/shared_common/CacheMap.cpp

```diff
@@ -2863,7 +2878,9 @@ SH_CacheMap::findROMClass(J9VMThread* currentThread, const char* path, Classpath
 #endif /* !defined(J9ZOS390) && !defined(AIXPPC) */
 	}
 
-	if (returnVal) { 
+	if (returnVal) {
+		/* Call updateROMSegmentList() to ensure that heapAlloc of the romClass segment is always updated to include the returned romClass */
+		updateROMSegmentList(currentThread, true);
```

Comment:
To minimize the impact of updateROMSegmentList(), we can make the following changes in updateROMSegmentList():
- it's only necessary to initialize classSegmentMutex when hasClassSegmentMutex is false
- don't bother having `J9JavaVM* vm`, it's only used once
- just use _cc and delete the _cacheletHead code

Reply:
Fixed. 
> just use _cc and delete the _cacheletHead code

_cc equals to _ccHead, instead of using _cc, I changed to use _ccHead. Also removed one unnecessary NULL check for cache inside updateROMSegmentList().